File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedToStringPrinted.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.ToStringPrintedRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.ClassNameUtils;
import com.ulyp.ui.util.StyledText;
import javafx.scene.Node;

import java.util.ArrayList;
import java.util.List;

import static com.ulyp.ui.util.CssClass.*;

public class RenderedToStringPrinted extends RenderedObject {

    public RenderedToStringPrinted(ToStringPrintedRepresentation representation, RenderSettings renderSettings) {
        super(representation.getType());

        String className = renderSettings.showTypes() ? representation.getType().getName() : ClassNameUtils.toSimpleName(representation.getType().getName());

        List<Node> nodes = new ArrayList<>();
        nodes.add(StyledText.of(className, CALL_TREE_TYPE_NAME));
        nodes.add(StyledText.of(": ", CALL_TREE_PLAIN_TEXT));
        nodes.add(RenderedObject.of(representation.getPrinted(), renderSettings));
        if (renderSettings.showTypes()) {
            nodes.add(StyledText.of("@", CALL_TREE_IDENTITY_REPR));
            nodes.add(StyledText.of(Integer.toHexString(representation.getIdentityHashCode()), CALL_TREE_IDENTITY_REPR));
        }

        super.getChildren().addAll(nodes);
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedToStringPrinted.kt
package com.ulyp.ui.renderers

import com.ulyp.core.printers.ToStringPrintedRepresentation
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.ClassNameUtils.toSimpleName
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of
import javafx.scene.Node

class RenderedToStringPrinted(representation: ToStringPrintedRepresentation, renderSettings: RenderSettings) : RenderedObject(representation.type) {

    init {
        val className =
            if (renderSettings.showTypes()) representation.type.name else toSimpleName(representation.type.name)
        val nodes: MutableList<Node> = ArrayList()
        nodes.add(of(className, CssClass.CALL_TREE_TYPE_NAME))
        nodes.add(of(": ", CssClass.CALL_TREE_NODE_SEPARATOR))
        nodes.add(of(representation.printed, renderSettings))
        if (renderSettings.showTypes()) {
            nodes.add(of("@", CssClass.CALL_TREE_IDENTITY_REPR))
            nodes.add(of(Integer.toHexString(representation.identityHashCode), CssClass.CALL_TREE_IDENTITY_REPR))
        }
        super.getChildren().addAll(nodes)
    }
}